alfresco-actuators/build.gradle

```diff
@@ -0,0 +1,93 @@
+plugins {
+    id 'java'
+    id 'eu.xenit.alfresco' version "1.0.1"
+    id 'eu.xenit.amp' version "1.0.1"
+    id 'org.hidetake.swagger.generator' version '2.11.0'
+    id "com.github.johnrengelman.shadow" version "5.1.0"
+}
+
+description = "Xenit webscripts to retrieve health of the application"
+group = 'eu.xenit.alfresco-actuators'
+
+// compile against alfresco 5.2
+ext {
+    alfrescoVersion='5.2.7'
+    servletApiVersion = '2.5'
+}
+
+tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    withJavadocJar()
+    withSourcesJar()
+}
+
+dependencies {
+    alfrescoProvided("org.alfresco:alfresco-repository:$alfrescoVersion")
+    alfrescoProvided("org.alfresco:alfresco-enterprise-repository:$alfrescoVersion")
+    alfrescoProvided("javax.servlet:servlet-api:$servletApiVersion")
+
+    // Add dependency for Swagger Codegen CLI
+    swaggerCodegen 'io.swagger:swagger-codegen-cli:2.2.3'
+
+    compileOnly 'io.swagger:swagger-annotations:1.5.6'
+    compileOnly 'javax.validation:validation-api:1.1.0.Final'
+    compile 'com.fasterxml.jackson.core:jackson-annotations:2.9.0'
+    compile("javax.annotation:javax.annotation-api:1.3.2")
+
+}
+
+swaggerSources {
+
+    model {
+        inputFile = file("src/main/resources/swagger.yaml")
+        code {
+            language = 'spring'
+            additionalProperties = [
+                    "modelPackage"     : "eu.xenit.actuators.model.gen",
+                    "serializableModel": "true"
+            ]
+            // Only generate model:
+            components = ['models']
+        }
+    }
+
+}
+
+processResources {
+    filesMatching("**/module.properties") {
```

Comment:
It's for the generated jar, to be able to use it as a simple module.

Reply:
The generated jar = the swagger spec ?

Reply:
No, the alfresco-actuators-0.0.1-SNAPSHOT.jar.
The jar can be used instead of the amp, as an alfrescoSM dependency. Due to the fact that it is deployed in WEB-INF/lib, module.properties and module-context.xml will be found and the jar will work correctly (see https://docs.alfresco.com/6.0/concepts/dev-extensions-packaging-techniques-jar-files.html).
The approach was used in alfred-telemetry, I thought that it was fine to use it, although it is not standard (then again, the xenit gradle plugin for SM is not standard).